packages/x-tree-view/src/internals/components/TreeItem/TreeItem.tsx

```diff
@@ -283,40 +218,33 @@ export const TreeItem = React.forwardRef(function TreeItem(
   function handleFocus(event: React.FocusEvent<HTMLLIElement>) {
     // DOM focus stays on the tree which manages focus with aria-activedescendant
     if (event.target === event.currentTarget) {
-      let rootElement: any;
```

Comment:
Having a method at the tree view level avoid to rely on the DOM structure and make the tree id an implementation detail at the item level.